server/images.go

```diff
@@ -56,19 +57,37 @@ type PromptVars struct {
 	First    bool
 }
 
-func (m *Model) Prompt(p PromptVars) (string, error) {
+// extractParts extracts the parts of the template before and after the {{.Response}} node.
+func extractParts(tmplStr string) (pre string, post string, err error) {
+	tmpl, err := template.New("").Parse(tmplStr)
+	if err != nil {
+		return "", "", err
+	}
+
+	var foundResponse bool
+
+	for _, node := range tmpl.Tree.Root.Nodes {
+		if node.Type() == parse.NodeAction && node.String() == "{{.Response}}" {
+			foundResponse = true
+		}
+		if !foundResponse {
+			pre += node.String()
+		} else {
+			post += node.String()
+		}
+	}
+
+	return pre, post, nil
+}
+
+func Prompt(promptTemplate string, p PromptVars) (string, error) {
```

Comment:
It's unclear to me why this is removing the receiver

Reply:
I removed it because it doesnt use any of the fields on the model, I can put the receiver back here if you prefer to have all the templating under the same object though

Reply:
It uses `model.Template` no? My main confusion is why `Prompt` doesn't have a receiver but `PreResponsePrompt` and `PostResponsePrompt` do.

Reply:
I think the image I sent in the other comment will help here too, but my mental `Prompt(template)` is re-used everywhere to apply our variable to a given template string. The template string can vary based on the place that is creating the prompt template.